lib/rules/utils/ast-utils.js

```diff
@@ -1104,7 +1104,6 @@ function isTopLevelExpressionStatement(node) {
 
 	return (
 		parent.type === "Program" ||
-		parent.type === "TSModuleBlock" ||
```

Comment:
I think we should keep this, so that checking TypeScript code works well with `{ "ignoreDirectives": true }`.